### ISSUE TITLE
OpusTrainer should disable early stopping until the final stage

### ISSUE THREAD

**Issue by gregtatum** [mozilla/firefox-translations-training#293]
While training one of my teachers in the ensemble started out very poorly, and was stopped due to early stopping before it could get to the later stages. Here this is the yellow graph of bad behavior.

Loss:
![image](https://github.com/mozilla/firefox-translations-training/assets/1588648/15afd8d1-8bdf-4e6f-a9e3-7daf84f3cb34)

chrF:
![image](https://github.com/mozilla/firefox-translations-training/assets/1588648/50376f3a-0145-4d7b-9eb3-a5b24191e3e7)

This is the in task group [PCOkERaaRtu6s6I7-xE5aA](https://firefox-ci-tc.services.mozilla.com/tasks/groups/PCOkERaaRtu6s6I7-xE5aA).

**Comment by gregtatum** [comment]
@eu9ene I'm thinking the work here is to have two different training steps. The first step would be to apply N-1 of the training schedule with `early-stopping` set to 0. Once that completes, we then would run the final stage with `early-stopping` being taken from the config.

**Comment by eu9ene** [comment]
> @eu9ene I'm thinking the work here is to have two different training steps. The first step would be to apply N-1 of the training schedule with `early-stopping` set to 0. Once that completes, we then would run the final stage with `early-stopping` being taken from the config.

Yes, to do that we should implement support of training parameters for each stage on the OpusTrainer side. However we used to train on the mixed dataset for 2 epochs with default early stopping (20) and it worked fine. So, we should investigate what changed here. Maybe in fact it did early stop sometimes but since we used a different task for finetuning it didn't affect it. I think the main issue might be just the proportion of the back-translated data + pre-training on the original. We might be able to fix it even without using different parameters for now. See #314.
